HWs.playground/Contents.swift

```diff
@@ -1,3 +1,201 @@
 import UIKit
 
-var greeting = "Hello, playground"
+//1) Постарайтесь больше практиковаться, повторите то, что есть в видео.
+
+//nil
+
+var date: String?
+
+date = "30.02.2017"
+
+//if let unwrappedDate = date {
+  //  print("Текущая дата \(unwrappedDate)")
+//} else {
+ //   print("Дата не была получена")
+//}
+
+//print(date!)
+
+//if date != nil {
+  //  print("Текущая дата \(date!)")
+//} else {
+  //  print("nil")
+//}
+
+let age: String = "20"
+
+if Int(age) != nil {
+    let ageInt = Int(age)!
+    ageInt
+}
+
+
+//2) Зайти и обязательно познакомиться с документацией. Выпишите в ДЗ 5 примеров
+
+let shortForm: Int? = Int("42")
+let longForm: Optional<Int> = Int("42")
+
+let number: Int? = Optional.some(42)
+let noNumber: Int? = Optional.none
+print(noNumber == nil)
+
+let imagePaths = ["star": "/glyphs/star.png",
+                  "portrait": "/images/content/portrait.jpg",
+                  "spacer": "/images/shared/spacer.gif"]
+
+if let starPath = imagePaths["star"] {
+    print("The star image is at '\(starPath)'")
+} else {
+    print("Couldn't find the star image")
+}
+
+if imagePaths["star"]?.hasSuffix(".png") == true {
+    print("The star image is in PNG format")
+}
+
+let defaultImagePath = "/images/default.png"
+let heartPath = imagePaths["heart"] ?? defaultImagePath
+print(heartPath)
+
+
+//3) Создать 5 констант со СТРОКАМИ которые содержат цифры и посчитайте их используя новые инструменты.
+
+let numOne = "1"
+let numTwo = "2"
+let numThree = "3"
+let numFour = "4"
+let numFive = "5"
+
+let arrayNum = [numOne, numTwo, numThree, numFour, numFive]
+var sum = Int()
+
+for string in arrayNum {
+    for char in string {
+        if let num = Int(String(char)) {
+            sum += num
+        }
+    }
+}
+
+print(sum)
+
+
+//4) Создать 3 константы со значением nil.
+
+let constOne: Int? = nil
+let constTwo: Double? = nil
+let constThree: Float? = nil
+
+
+//5) Создать 5 опциональных типов констант и установите им значения.
+
+let firstConst: Int? = 1
+let secondConst: Double? = 1.1
+let thirdConst: Float? = 1.11
+let fourthConst: String = "Hello"
+let fifthConts: Bool? = false
+
+
+//6) Перечислите способы извлечения опционала с примерами (guard не использовать)
+
+// -  Force unwrapping.
+
+let email: String? = "angelovsv@icloud.com"
+
+print(email!)
+
+// - Implicity unwrapping.
+
+let optionalString: String? = "Good Morning!"
+let optionalStringTwo: String! = optionalString!
+
+// - Optional binding.
+
+let optionalUserName: String? = "Sergey"
+
+if let userName = optionalUserName {
+    print("Good morning \(userName)!")
```

Comment:
```suggestion
    print("Good morning \(userName)!")
 ```

тут же вроде не нужен !